core/src/main/scala/kafka/network/RequestConvertToJson.scala

```diff
@@ -0,0 +1,343 @@
+      case res: ListPartitionReassignmentsResponse => ListPartitionReassignmentsResponseDataJsonConverter.write(res.data(), version)
+      case res: MetadataResponse => MetadataResponseDataJsonConverter.write(res.data(), version)
+      case res: OffsetCommitResponse => OffsetCommitResponseDataJsonConverter.write(res.data(), version)
+      case res: OffsetDeleteResponse => OffsetDeleteResponseDataJsonConverter.write(res.data, version)
+      case res: OffsetFetchResponse => OffsetFetchResponseDataJsonConverter.write(res.data, version)
+      case res: OffsetsForLeaderEpochResponse => offsetsForLeaderEpochResponseNode(res, version)
+      case res: ProduceResponse => produceResponseNode(res, version)
+      case res: RenewDelegationTokenResponse => RenewDelegationTokenResponseDataJsonConverter.write(res.data(), version)
+      case res: SaslAuthenticateResponse => SaslAuthenticateResponseDataJsonConverter.write(res.data(), version)
+      case res: SaslHandshakeResponse => SaslHandshakeResponseDataJsonConverter.write(res.data(), version)
+      case res: StopReplicaResponse => StopReplicaResponseDataJsonConverter.write(res.data(), version)
+      case res: SyncGroupResponse => SyncGroupResponseDataJsonConverter.write(res.data, version)
+      case res: TxnOffsetCommitResponse => TxnOffsetCommitResponseDataJsonConverter.write(res.data, version)
+      case res: UpdateFeaturesResponse => UpdateFeaturesResponseDataJsonConverter.write(res.data(), version)
+      case res: UpdateMetadataResponse => UpdateMetadataResponseDataJsonConverter.write(res.data(), version)
+      case res: WriteTxnMarkersResponse => WriteTxnMarkersResponseDataJsonConverter.write(res.data, version)
+      case res: VoteResponse => VoteResponseDataJsonConverter.write(res.data, version)
+    }
+  }
+
+  def requestHeaderNode(header: RequestHeader): JsonNode = {
+    val node = RequestHeaderDataJsonConverter.write(header.data(), header.headerVersion()).asInstanceOf[ObjectNode]
+    node.set("requestApiKey", new TextNode(header.apiKey.toString))
+    node
+  }
+
+  def requestDescMetrics(header: RequestHeader, res: Response, req: AbstractRequest,
+                  context: RequestContext, session: Session, verbose: Boolean,
+                  totalTimeMs: Double, requestQueueTimeMs: Double, apiLocalTimeMs: Double,
+                  apiRemoteTimeMs: Double, apiThrottleTimeMs: Long, responseQueueTimeMs: Double,
+                  responseSendTimeMs: Double, temporaryMemoryBytes: Long,
+                  messageConversionsTimeMs: Double): JsonNode = {
+    val node = new ObjectNode(JsonNodeFactory.instance)
+    node.set("requestHeader", requestHeaderNode(header))
+    node.set("request", request(req, verbose))
+    node.set("response", res.responseLog.getOrElse(new TextNode("")))
+    node.set("connection", new TextNode(context.connectionId))
+    node.set("totalTime", new DoubleNode(totalTimeMs))
```

Comment:
I was thinking a bit more about the time fields, and I think we should add `Ms` to the end of all of them. What do you think?

Reply:
I'm for it, that would be more descriptive.